FlowCrypt/Controllers/SideMenu/Main/SideMenuNavigationController.swift

```diff
@@ -114,38 +114,39 @@ extension SideMenuNavigationController {
                 }
                 let emailKeyManagerApi = EmailKeyManagerApi(clientConfiguration: configuration)
                 let idToken = try await IdTokenUtils.getIdToken(userEmail: context.user.email)
-                let keydDetails = try await emailKeyManagerApi.getPrivateKeys(idToken: idToken)
+                let keyDetails = try await emailKeyManagerApi.getPrivateKeys(idToken: idToken)
                 let localKeys = try context.encryptedStorage.getKeypairs(by: context.user.email)
-                var savedPassPhrase = localKeys.first(where: { $0.passphrase != nil })?.passphrase
-                var isKeyUpdated = false
-                for keyDetail in keydDetails {
-                    guard let savedLocalKey = localKeys.first(where: { $0.primaryFingerprint == keyDetail.primaryFingerprint }) else {
-                        // No keys found in local. Add it
-                        savedPassPhrase = try await saveKeyToLocal(
-                            context: context,
-                            keyDetail: keyDetail,
-                            passPhrase: savedPassPhrase,
-                            isNewKey: true
-                        )
-                        isKeyUpdated = true
-                        continue
-                    }
-                    guard let lastModified = keyDetail.lastModified else {
-                        continue
-                    }
-                    // Key exists in local. Check if saved key is outdated by checking lastModified and update if needed
-                    if savedLocalKey.lastModified ?? 0 < lastModified {
-                        savedPassPhrase = try await saveKeyToLocal(
-                            context: context,
-                            keyDetail: keyDetail,
-                            passPhrase: savedPassPhrase
-                        )
-                        isKeyUpdated = true
+
+                var passPhrase = localKeys.first(where: { $0.passphrase != nil })?.passphrase
+                // If this is called when starting the app, then it doesn't make much difference
+                // but conceptually it would be better to look pass phrase both in memory and storage
+                if passPhrase == nil {
+                    passPhrase = try context.passPhraseService.getPassPhrases().first(where: { $0.value.isNotEmpty })?.value
+                }
+
+                // second parameter indicates whether key detail is new key or not
+                var keysToUpdate: [(KeyDetails, Bool)] = []
+                for keyDetail in keyDetails {
+                    if let savedLocalKey = localKeys.first(where: { $0.primaryFingerprint == keyDetail.primaryFingerprint }) {
+                        if savedLocalKey.lastModified ?? 0 < keyDetail.lastModified ?? 0 {
+                            keysToUpdate.append((keyDetail, false))
+                        }
```

Comment:
I saw your comment. If you would want to prevent a nil `lastModified` from being saved to storage (I think that's a good idea), then let's take this a step further: 
 - make `lastModified` in storage not optional. Do a migration and in the migration, remove keys that are missing `lastModified`
 - add a `guard` here to ensure last modified is present. If not present, instead of skipping the key, throw as if the key was invalid.

Reply:
I set Realm `lastModified` to be nullable because api response type is also nullable.
![image](https://user-images.githubusercontent.com/77344191/165728110-33ea2765-3b01-45dd-a11e-a92874e88c5a.png)

Reply:
Sure. But keys that don't have information in them when they were last modified are useless keys (they lack pgp self-signatures).

Therefore, I propose (inspired by your original code) that the storage should be not optional, and that we ensure that only keys that have this are saved to storage.

Reply:
ok

Reply:
@tomholub one more question regarding `lastModified`.
For all beta users, key `lastModified` value is missing and we have to delete all keys for all users which means they will lose all infos related to key.
Is this fine?

> make lastModified in storage not optional. Do a migration and in the migration, remove keys that are missing lastModified

Reply:
No, you have to parse the existing keys during the db migration, and figure out what their `lastModified` is.

Only when you parse it and it returns a nil `lastModified` then it's useless key anyway and can be deleted.

Reply:
ok

Reply:
@sosnovsky Do you have any idea how to pass asynchronous function in `enumerateObjects`?
I haven't found solution yet.

![image](https://user-images.githubusercontent.com/77344191/165765008-2ba7064f-f3c1-45e1-9852-938ea974d88f.png)

Reply:
@sosnovsky Ohh, yes So simple solution. I thought it in too difficult way.

Reply:
Just need to test if `Task` works well for Realm migrations

Reply:
Yeah

Reply:
as we expected, it doesn't work well.
After it gets parsed result I can't set newObject detail because migration block is already finished and displays this error.
Finding solution now though.
`Object has been deleted or invalidated `

Reply:
As I checked deeply we can't accomplish this with default Realm migration schema.
I think only way to implement this is like this.

1. Set lastModified default value to 0.
2. When app starts check if isLastModifiedUpdated (setting value which might be saved in UserDefaults or something like that)is true and perform update `lastModified` logic before any other operations.
3. Set `isLastModifiedUpdated` to true.

@tomholub, @sosnovsky Please confirm

Reply:
I don't have any better idea. Sounds ok to me. Add a comment to remind us to remove it a few versions later.

Reply:
ok